### DISCUSSION TITLE
How to install lefthook using python?

### DISCUSSION THREAD

**Discussion by nktrnv** [evilmartians/lefthook#620] [Q&A] [answered]
I am trying to install lefthook using the `pip install lefthook` command. I use Windows. Do I need to install Go? Anyway, I decided to install Go, because it is not working without it. Then I run the `lefthook` command and get lefthook executable not found. If I run the 'python -m lefthook` command, the following error occurs:
```
# lefthook not found, trying to install
go: go.mod file not found in current directory or any parent directory.
        'go get' is no longer supported outside a module.
        To build and install a command, use 'go install' with a version,
        like 'go install example.com/cmd@latest'
        For more information, see https://golang.org/doc/go-get-install-deprecation
        or run 'go help get' or 'go help install'.
# brew not found
Traceback (most recent call last):
...
```
Everything only worked when I installed `lefthook` using `go install`. Can I install lefthook locally in python virtual environment without the need to install Go? I use `pre-commit` and I easily can install it as a project dependency and use it in virtual environment. I'm sorry if I understand something wrong!

**Comment by mrexox** [comment, marked as answer]
Hey! `lefthook` package for pip is a third party package and it looks like it was [abandoned](https://github.com/orsinium-archive/lefthook/tree/master). There is no way right now to install lefthook using pip, but you can probably try [Winget](https://github.com/evilmartians/lefthook/blob/master/docs/install.md#winget).

To be honest, it was a surprise for me that the Python package is actually just tries to install lefthook [using other package mangers](https://github.com/orsinium-archive/lefthook/blob/master/lefthook/_core.py#L6-L11). So, I don't think it's a good idea to use it even it it worked :)

**Reply by nktrnv** [reply]
OK, thanks for the feedback!

**Reply by sche34** [reply]
Looks like they started updating it again in october 2024. Should be safe to use now
